File: smart-family-dmt/DMTServer/model/profileDeviceAccessResponse.py
class ProfileDeviceAccess:
    device_id: int
    node_id: int
    access: bool
    location: str
    profile_id: int
    profile_name: str
    profile_role: str

    def __init__(self, device_id, node_id, access, location, profile_id, profile_name, profile_role):
        self.device_id = device_id
        self.node_id = node_id
        self.access = access
        self.location = location
        self.profile_id = profile_id
        self.profile_name = profile_name
        self.profile_role = profile_role


class ProfileDeviceAccessResponse:
    pda_list: [ProfileDeviceAccess] = []

    def __init__(self, pda_list):
        self.pda_list = pda_list

File: smart-family-dmt/SmartF-DMT/SmartF-DMT/Models/ProfileDeviceAccessResponse.swift
//
//  ProfileDeviceAccessResponse.swift
//  SmartF-DMT
//
//  Created by Wei Zhang on 4/22/24.
//

import Foundation


struct ProfileDeviceAccessResponse: Codable {
  let statusCode: Int
  let body: ProfileDeviceAccessResponseBody
}

struct ProfileDeviceAccessResponseBody: Codable {
  var profileDeviceAccessList: [ProfileDeviceAccess]
  enum CodingKeys: String, CodingKey {
    case profileDeviceAccessList="pda_list"
  }
}

struct ProfileDeviceAccess: Codable {
  let id = UUID()
  let deviceId: Int
  let nodeId: Int
  let access: Bool
  let location: String
  let profileId: Int
  let profileName: String
  let profileRole: String
  enum CodingKeys: String, CodingKey {
    case deviceId="device_id",
         nodeId = "node_id",
         access,
         location,
         profileId="profile_id",
         profileName="profile_name",
         profileRole="profile_role"
  }
}
